src/plugins/memdump/memdump.h

```diff
@@ -177,6 +178,27 @@ typedef struct target_config_entry
     std::string function_name;
 } target_config_entry_t;
 
+// type of a pointer residing on stack
+enum sptr_type_t
+{
+    ERROR,   // problem with stack inspection
+    MAIN,    // pointer to a main module
+    LINKED,  // pointer to some linked DLL module
+    UNLINKED // pointer to some non-legit memory
+};
+
+sptr_type_t check_module_linked_wow(drakvuf_t drakvuf,
+                                    vmi_instance_t vmi,
+                                    memdump* plugin,
+                                    drakvuf_trap_info_t* info,
+                                    addr_t dll_base);
+
+sptr_type_t check_module_linked(drakvuf_t drakvuf,
+                                vmi_instance_t vmi,
+                                memdump* plugin,
+                                drakvuf_trap_info_t* info,
+                                addr_t dll_base);
```

Comment:
All these additions look to me like they belong in a `private.h` and not in the `memdump.h` which is included by `plugins.h`. There are a bunch of other stuff in `memdump.h` that should also be instead in this `private.h`, like `target_config_entry_t` etc

Reply:
Actually `target_config_entry_t` is a part of the definition of class `memdump`, so I'm not sure if this could be moved. I guess the compiler will complain.

```cpp
class memdump: public pluginex
{
/* ... */
    std::vector<target_config_entry_t> wanted_hooks;
```

Reply:
Even in that case `memdump.h` should only have a forward declaration of the type. Pretty much only stuff that needs to be accessible/defined outside the plugin itself lives in that header, everything else should be in a `private.h`

Reply:
most stuff moved to `private.h`